lib/cocoapods/generate/installer.rb

```diff
@@ -248,9 +256,10 @@ def add_test_spec_schemes_to_app_scheme(installer, app_project)
           .flat_map(&:test_native_targets)
           .group_by(&:platform_name)
 
+        workspace_path = install_directory + "#{spec.name}.xcworkspace"
         Xcodeproj::Plist.write_to_path(
           { 'IDEWorkspaceSharedSettings_AutocreateContextsIfNeeded' => false },
-          app_project.path.sub_ext('.xcworkspace').join('xcshareddata').tap(&:mkpath).join('WorkspaceSettings.xcsettings')
+          workspace_path.join('xcshareddata').tap(&:mkpath).join('WorkspaceSettings.xcsettings')
```

Comment:
this was a bug I introduced in the previous PR. The app project path can be "PodSample" and this was creating an extra and unused folder.